source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,257 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
```

Comment:
Those platforms also need to differ.

Reply:
You mean it's not possible for build platform to also be the host platform?

Reply:
That is possible, but that is native compilation, usually. However, you can also do “native” cross compilation (like exherbo) which is also [a goal for nixpkgs I subscribe to](https://github.com/NixOS/nixpkgs/issues/21471).

Since cross compilation is the more generic concept, it is probably hard disambiguate — in a way native compilation is a special case of cross compilation. Following that train of thought you could talk about cross compilation as soon as your compiler has a notion of a target platform.

I think my original suggestion may not be the best one, but the initial paragraph reads to me as if it is missing something, maybe?

Reply:
How about a second paragraph that says:

> Cross-compilation is generalized from native compilation, where host and build platforms are the same.

Reply:
Why not!